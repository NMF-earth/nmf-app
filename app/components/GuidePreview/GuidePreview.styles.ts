import { StyleSheet } from "react-native";

export default StyleSheet.create({
  title: {
    paddingTop: 28,
    paddingBottom: 16,
    textAlign: "center"
  },
  listItem: {
    flex: 1,
    padding: 16
  },
  button: {
    marginTop: 20,
    alignSelf: "center"
  }
});
