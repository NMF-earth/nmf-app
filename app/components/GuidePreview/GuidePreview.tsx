import React from "react";
import { FlatList, TouchableOpacity } from "react-native";
import { t } from "../../utils";

import Text from "../../components/Text";
import Button from "../../components/Button";
import styles from "./GuidePreview.styles";
import { Guide } from "../../types/common-types";

interface Props {
  title: string;
  listItems: Guide[];
  onPressItem: (item: Guide) => void;
  onPressSeeAll: () => void;
}

class GuidePreview extends React.Component<Props> {
  renderItem = ({ item }: { item: Guide }) => (
    <TouchableOpacity
      onPress={() => this.props.onPressItem(item)}
      style={styles.listItem}
    >
      <Text.Tertiary>{item.title}</Text.Tertiary>
    </TouchableOpacity>
  );

  renderHeader = () => (
    <Text.H3 style={styles.title}>{this.props.title}</Text.H3>
  );

  renderFooter = () => (
    <Button.Secondary
      style={styles.button}
      onPress={() => this.props.onPressSeeAll()}
      textType={"Secondary"}
    >
      <Text.Secondary bold green center>
        {t("GUIDE_PREVIEW_SEE_ALL")}
      </Text.Secondary>
    </Button.Secondary>
  );

  render() {
    return (
      <FlatList
        ListHeaderComponent={this.renderHeader}
        data={this.props.listItems}
        renderItem={this.renderItem}
        ListFooterComponent={this.renderFooter}
      />
    );
  }
}

export default GuidePreview;
