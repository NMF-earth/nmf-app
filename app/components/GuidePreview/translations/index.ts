import en from "./en";
import fr from "./fr";
import de from "./de";
import sv from "./sv";

interface TranslationKeys {
  GUIDE_PREVIEW_SEE_ALL: string;
}

export { en, fr, de, sv, TranslationKeys };
