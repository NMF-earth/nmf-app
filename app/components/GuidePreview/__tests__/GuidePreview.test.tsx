import React from "react";
import renderer from "react-test-renderer";
import GuidePreview from "../";

it("renders correctly TabbedView", () => {
  const tree = renderer.create(<GuidePreview title="Foo" listItems={[]}  onPress={() => {}}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
