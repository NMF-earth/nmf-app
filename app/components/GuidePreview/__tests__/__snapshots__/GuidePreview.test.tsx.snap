// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly TabbedView 1`] = `
Array [
  <Text.H3
    style={
      Object {
        "paddingBottom": 16,
        "paddingTop": 28,
        "textAlign": "center",
      }
    }
  >
    Foo
  </Text.H3>,
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>,
  <Button.Secondary
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "marginTop": 20,
      }
    }
    textType="Secondary"
  >
    <Text.Primary
      bold={true}
      center={true}
      green={true}
    >
      See All
    </Text.Primary>
  </Button.Secondary>,
]
`;
